CHANGELOG.md

```diff
@@ -4,6 +4,182 @@ All notable changes to Router will be documented in this file.
+
+By [@garypen](https://github.com/garypen) in https://github.com/apollographql/router/pull/2627
+
+### Activate TLS for Redis cluster connections ([Issue #2332](https://github.com/apollographql/router/issues/2332))
+
+This adds support for TLS connections in Redis Cluster mode, by applying it when the URLs use the `rediss` schema.
+
+By [@Geaal](https://github.com/Geal) in https://github.com/apollographql/router/pull/2605
+
+### Make `terminationGracePeriodSeconds` property configurable in the Helm chart
+
+The `terminationGracePeriodSeconds` property is now configurable on the `Deployment` object in the Helm chart.
+
+This can be useful when adjusting the default timeout values for the router, and should always be a value slightly bigger than the timeout in order to ensure no requests are closed prematurely on shutdown.
+
+By [@Meemaw](https://github.com/Meemaw) in https://github.com/apollographql/router/pull/2582
+
+## 🐛 Fixes
+
+### Properly emit histograms metrics via OpenTelemetry ([Issue #2393](https://github.com/apollographql/router/issues/2493))
+
+With the "inexpensive" metrics selector, histograms are only reported as gauges which caused them to be incorrectly interpreted when reaching Datadog
+
+By [@Geal](https://github.com/geal) in https://github.com/apollographql/router/pull/2564
+
+### Revisit Open Telemetry integration ([Issue #1812](https://github.com/apollographql/router/issues/1812), [Issue #2359](https://github.com/apollographql/router/issues/2359), [Issue #2338](https://github.com/apollographql/router/issues/2338), [Issue #2113](https://github.com/apollographql/router/issues/2113), [Issue #2113](https://github.com/apollographql/router/issues/2113))
+
+There were several issues with the existing OpenTelemetry integration in the router which we are happy to have resolved with this re-factoring:
+
+- Metrics would stop working after a schema or config update.
+- Telemetry config could **not** be changed at runtime, instead requiring a full restart of the router.
+- Logging format would vary depending on where the log statement existed in the code.
+- On shutdown, the following message occurred frequently:
+
+  ```
+  OpenTelemetry trace error occurred: cannot send span to the batch span processor because the channel is closed
+  ```
+
+- And worst of all, it had a tendency to leak memory.
+
+We have corrected these by re-visiting the way we integrate with OpenTelemetry and the supporting tracing packages and brought our usage to align with new best-practices.
+
+In addition, the testing coverage for telemetry in general has been significantly improved.  For more details of what changed and why take a look at https://github.com/apollographql/router/pull/2358.
+
+By [@bryncooke](https://github.com/bryncooke) and [@geal](https://github.com/geal) and [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/2358
```

Comment:
```suggestion
There were several issues with the existing OpenTelemetry integration in the Router which we are happy to have resolved with this re-factoring:

- Metrics would stop working after a schema or config update.
- Telemetry config could **not** be changed at runtime, instead requiring a full restart of the router.
- Logging format would vary depending on where the log statement existed in the code.
- On shutdown, the following message occurred frequently:

  ```
  OpenTelemetry trace error occurred: cannot send span to the batch span processor because the channel is closed
  ```

- And worst of all, it had a tendency to leak memory.

We have corrected these by re-visiting the way we integrate with OpenTelemetry and the supporting tracing packages. The new implementation brings our usage in line with new best-practices.

In addition, the testing coverage for telemetry in general has been significantly improved.  For more details of what changed and why take a look at https://github.com/apollographql/router/pull/2358.

By [@bryncooke](https://github.com/bryncooke) and [@geal](https://github.com/geal) and [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/2358
```